### PR TITLE
Feature/event listener

### PR DESCRIPTION
**Pull request by JulienTD**
## Changes
 - Added an event listener
 - Added event containers (FalconButtonEvent, FalconKeyboardEvent, FalconWheelEvent)

### CODE REVIEW

**Review by ImOverlord — CHANGES_REQUESTED** [review]
Add a gitignore to prevent pushing build files

File: src/FalconEye.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        this.eventListener.listen();
+        this.hasStarted = true;
+    }
+
+    public getAPIKey(): string {
+        return this.apiKey;
+    }
+}
+
+let fe: FalconEye = new FalconEye("adkfdsnfsnfdjnfdsjfsfssfd");
```
Comment (ImOverlord) [inline src/FalconEye.ts:34]: You don't need to specify the type here it should be implied

**Review by ImOverlord — COMMENTED** [review]
Maybe add an example, so we can quickly test your feature

File: src/FalconEye.ts (hunk 1 above)
Comment (JulienTD) [reply src/FalconEye.ts:34]: This line is just a test. On the release, it'll be removed, but I can remove it now

**Comment by JulienTD** [comment]
> Maybe add an example, so we can quickly test your feature

To test it, you just need to add the script to an html file

**Comment by JulienTD** [comment]
Ok from the last commit, you need to run `npm run build` to build the project and `npm run server` to start the server on port 3000

File: src/ConfigProfile.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Persistent } from "../node_modules/persistent-typescript/build/Persistent";
+
+@Persistent()
+export class ConfigProfile {
+
+    private apiKey: string;
```
Comment (ImOverlord) [inline src/ConfigProfile.ts]: Maybe use native Ts set and get?

File: src/event/FalconButtonEvent.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { FalconEvent } from "./FalconEvent";
+
+export enum ButtonType {
```
Comment (ImOverlord) [inline src/event/FalconButtonEvent.ts]: We Should seperate enums from class

File: src/event/FalconEvent.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+export class FalconEvent {
```
Comment (ImOverlord) [inline src/event/FalconEvent.ts]: We could export this class as default

**Review by ImOverlord — COMMENTED** [review]
Everything looks good, made a few a comment before we can merge.

File: src/event/FalconEvent.ts (hunk 4 above)
Comment (JulienTD) [reply src/event/FalconEvent.ts]: We can try export default, but if it makes too many maintainability issues we will have to remove it

File: src/event/FalconButtonEvent.ts (hunk 3 above)
Comment (JulienTD) [reply src/event/FalconButtonEvent.ts]: Yes you're right, it's done

File: src/ConfigProfile.ts (hunk 2 above)
Comment (JulienTD) [reply src/ConfigProfile.ts]: Yes I didn't know it existed, its added

**Review by ImOverlord — APPROVED** [review]

**Pull request merged**